crates/bevy_tasks/src/wasm_task.rs

```diff
@@ -0,0 +1,82 @@
+use std::{
+    any::Any,
+    future::{Future, IntoFuture},
+    panic::{AssertUnwindSafe, UnwindSafe},
+    pin::Pin,
+    task::Poll,
+};
+
+use futures_channel::oneshot;
+
+/// Wraps an asynchronous task, a spawned future.
+///
+/// Tasks are also futures themselves and yield the output of the spawned future.
+#[derive(Debug)]
+pub struct Task<T>(oneshot::Receiver<Result<T, Panic>>);
+
+impl<T: 'static> Task<T> {
+    pub(crate) fn wrap_future(future: impl Future<Output = T> + 'static) -> Self {
+        let (sender, receiver) = oneshot::channel();
+        wasm_bindgen_futures::spawn_local(async move {
+            // Catch any panics that occur when polling the future so they can
+            // be propagated back to the task handle.
+            let value = CatchUnwind(AssertUnwindSafe(future)).await;
+            let _ = sender.send(value);
+        });
+        Self(receiver.into_future())
+    }
+
+    /// When building for wasm, this method has no effect.
+    /// This is only included for feature parity with other platforms.
+    pub fn detach(self) {}
+
+    /// Requests a task to be cancelled and returns a future that suspends until it completes.
+    /// Returns the output of the future if it has already completed.
+    ///
+    /// # Implementation
+    ///
+    /// When building for wasm, it is not possible to cancel tasks which means this is the same
+    /// as just awaiting the task. This method is only included for feature parity with other platforms.
+    pub async fn cancel(self) -> Option<T> {
+        match self.0.await {
+            Ok(Ok(value)) => Some(value),
+            Err(_) => None,
+            Ok(Err(panic)) => {
+                // drop this to prevent the panic panic from resuming the panic on drop.
```

Comment:
This sentence is very confusing to me

Reply:
Probably because it's nonsense haha. It's mean to say "panic payload", not panic panic